File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        
        int left=0;
        int right=heights.length-1;
        int maxWater = Math.min(heights[left],heights[right])*(right-left);
        while(left<right){
            int water = Math.min(heights[left],heights[right])*(right-left);
            if(water>maxWater){
                maxWater=water;
            }
            if(heights[left]<heights[right]){
                left++;
            }
            else{
                if(heights[left]>heights[right]){
                    right--;
                }
                else{
                    left++;
                }
            }


        }
        return maxWater;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left = 0
        right = len(nums) - 1
        middle = (left+right)//2
        while left<right:
            if nums[middle]>nums[middle+1]:
                return nums[middle+1]
            elif nums[middle]>nums[left]:
                left = middle
                middle = (left+right)//2
            else:
                right = middle
                middle = (left+right)//2
        return nums[0]